packages/react-utilities/src/hooks/useEvent.ts

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+type UseEvent = {
+  /**
+   * The element to listen to.
+   */
+  element?: Element | Window | Document;
+
+  /**
+   * The event type to watch.
+   */
+  type: string;
+
+  /**
+   * The callback to call when the eventName is triggered.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  callback: (ev: any) => void;
+
+  /**
+   * Whether the events should be dispatched to the callback first before other EventTargets.
+   */
+  useCapture?: boolean;
+
+  /**
+   * Whether event should be disabled and removed
+   */
+  disabled: boolean;
+};
+
+/**
+ * Hook used to add and remove a event from an element.
+ *
+ * @param element - The element to listen to
+ * @param type - The event to watch
+ * @param callback - The callback to call when the eventName is triggered
+ * @param useCapture - Whether the events should be dispatched to the callback first before other EventTargets
+ * @param disabled - Whether event should be disabled and removed
```

Comment:
I think it might also be better for this one to follow the signature of `addEventListener` and be a function with multiple parameters. `disabled` can probably be inferred from whether the other parameters are provided.